src/middleware/indexer.js

```diff
@@ -162,18 +174,67 @@ const findLikelyNFTs = async (ctx) => {
             and action_kind = 'FUNCTION_CALL'
             and args->>'args_json' is not null
             and args->>'method_name' like 'nft_%'
+            and receipt_included_in_block_timestamp <= $2
+            and receipt_included_in_block_timestamp > $3
     `;
 
     const ownershipChangeEvents = `
         select distinct emitted_by_contract_account_id as receiver_account_id 
         from assets__non_fungible_token_events
         where token_new_owner_account_id = $1
+            and emitted_at_block_timestamp <= $2
```

Comment:
Same comment here re: indexes. If we can append the timestamp to the existing index that's great, otherwise let's at least get some assurance that the change in performance is negligible.